scipy/stats/tests/test_stats.py

```diff
@@ -3793,6 +3794,28 @@ def test_ttest_rel():
     assert_raises(ValueError, stats.ttest_rel, x.reshape((8, 3)),
                   x.reshape((2, 3, 4)))
 
+    # Convert from two-sided p-values to one sided using T result data.
+    def convert(t, p, alt):
+        if (t < 0 and alt == "less") or (t > 0 and alt == "greater"):
+            return p / 2
+        return 1 - (p / 2)
+    converter = np.vectorize(convert)
+
+    rvs1_2D[:, 20:30] = np.nan
+    rvs2_2D[:, 15:25] = np.nan
```

Comment:
These are fine, but thoughts for the future.
* randomize the `nan` patterns
* test along each axis (`@parametrize`)
* test more than two dimensional arrays. I've seen functions that work for two dimensions, but not three or more.
* since `ttest_rel` without `nan`s is already tested for accuracy, test against that (i.e. loop over it, manually stripping out the `nan`s from each axis-slice)

Let's not change these, though. What is here is pretty consistent with the rest, and we need some infrastructure (e.g. gh-13312 ) to make some of these suggestions easy to implement.

Reply:
> These are fine, but thoughts for the future.
> 
> * randomize the `nan` patterns
> 
> * test along each axis (`@parametrize`)
> 
> * test more than two dimensional arrays. I've seen functions that work for two dimensions, but not three or more.
> 
> * since `ttest_rel` without `nan`s is already tested for accuracy, test against that (i.e. loop over it, manually stripping out the `nan`s from each axis-slice)

OK. All these suggestions sound good. I will do these from now on.

Reply:
They might be overkill. And until gh-gh-13312 gives an easy way of creating a reference implementation of `nan_policy` over multiple dimensions, it's not worth the trouble. So just thinking ahead.